File: frontendbuddyfinv/src/models/DetalleVentaResponseDTO.js
export default class DetalleVentaResponseDTO {
  constructor({ idDetalleVenta = null, productoId = null, productoNombre = '', cantidad = 0, precioUnitario = 0, subtotal = 0 } = {}) {
    this.idDetalleVenta = idDetalleVenta;
    this.productoId = productoId;
    this.productoNombre = productoNombre;
    this.cantidad = cantidad;
    this.precioUnitario = precioUnitario;
    this.subtotal = subtotal;
  }

  static fromApi(obj = {}) {
    return new DetalleVentaResponseDTO({
      idDetalleVenta: obj.idDetalleVenta,
      productoId: obj.productoId,
      productoNombre: obj.productoNombre,
      cantidad: obj.cantidad,
      precioUnitario: obj.precioUnitario,
      subtotal: obj.subtotal
    });
  }
}

File: backendbuddyfinv/src/main/java/com/es/backendbuddyfinv/dto/DetalleVentaResponseDTO.java
package com.es.backendbuddyfinv.dto;

import lombok.Data;



@Data

public class DetalleVentaResponseDTO {
        private Long idDetalleVenta;
    private Long productoId;
    private String productoNombre;
    private Integer cantidad;
    private Double precioUnitario;
    private Double subtotal;
    // getters / setters by lombok

    
}
